src/hera/workflows/script.py

```diff
@@ -379,6 +380,7 @@ def _get_parameters_from_callable(source: Callable) -> List[Parameter]:
             default = MISSING
 
         param = Parameter(name=p.name, default=default)
+        add_metadata_from_type(param, p.annotation)
```

Comment:
I wasn't sure if we wanted to populate param.enum from the Literal metadata in the non-experimental path, or just when the `script_annotations` experimental feature is enabled, so I put this change into a separate commit to make it easy to revert.

Reply:
I would prefer to do it just for script annotations for now

Reply:
That does mean that the generated YAML for a non-Annotated script will change depending on whether an unrelated experimental feature is enabled though.